docs/metrics.md

```diff
@@ -0,0 +1,97 @@
+
+### Histograms (`NewHistogram`)
+Tracks aggregates (count, min, max, mean, percentiles, etc) for samples
+expressed in arbitrary base unit.
+
+### Timers (`NewTimer`)
+Tracks aggregates (count, min, max, mean, percentiles, etc) of samples expressed in units of time.
+
+### Meters (`NewMeter`)
+Tracks aggregates (count, min, max, mean, etc),  rate (1m, 5m, 15m) for samples
+expressed in base unit.
+
+
+Metric name                       | Type      | Description
+--------------------------------  | --------  | --------------------
+bucket.snap.merge                 | timer     | time to merge two buckets
+bucket.batch.insert               | meter     | number of objects added per batch
+bucket.batch.add                  | timer     | time to add a batch
+bucket.memory.shared              | counter   | number of buckets referenced (excluding publish queue)
+history.catchup.failure           | meter     | catchup failed
+scp.sync.lost                     | meter     | validator lost sync
+scp.envelope.emit                 | meter     | SCP message sent
+scp.envelope.receive              | meter     | SCP message received
+scp.memory.cumulative-statements  | counter   | number of known SCP statements known
+herder.pending-txs.age0           | counter   | number of gen0 pending transactions
+herder.pending-txs.age1           | counter   | number of gen1 pending transactions
+herder.pending-txs.age2           | counter   | number of gen2 pending transactions
+herder.pending-txs.age3           | counter   | number of gen3 pending transactions
+scp.envelope.sign                 | meter     | envelope signed
+scp.envelope.validsig             | meter     | envelope signature verified
+scp.envelope.invalidsig           | meter     | envelope failed signature verification
+scp.value.valid                   | meter     | SCP value is valid
+scp.value.invalid                 | meter     | SCP value is invalid
+scp.nomination.combinecandidates  | meter     | number of candidates per call
+scp.timing.nominated              | timer     | time spent in nomination
+scp.timing.externalized           | timer     | time spent in ballot protocol
+scp.pending.processed             | counter   | number of already processed envelopes
+scp.pending.discarded             | counter   | number of discarded envelopes
+scp.pending.fetching              | counter   | number of incomplete envelopes
+scp.pending.ready                 | counter   | number of envelopes ready to process
+history.publish.success           | meter     | published completed successfuly
+history.publish.failure           | meter     | published failed
+history.download-<X>.success      | meter     | download of <X> completed successfuly
+history.download-<X>.failure      | meter     | download of <X> failed
+history.verify-<X>.success        | meter     | verification of <X> succeeded
+history.verify-<X>.failure        | meter     | verification of <X> failed
```

Comment:
Note: in current master, `history.verify-ledger.*` metrics have finer granularity, but there's a change in https://github.com/stellar/stellar-core/pull/1711 to use `history.verify-ledger.success` and `history.verify-ledger.failure` only.